### PR TITLE
test: gateway full coverage

### PR DESCRIPTION
**Pull request by Foivos**
[AXE-5568]

[AXE-5568]: https://axelarnetwork.atlassian.net/browse/AXE-5568?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: move/axelar_gateway/sources/gateway.move [hunk 1]
```diff
@@ -469,37 +564,474 @@ fun test_peel_proof_no_remaining_data() {
...
+    let data_hash = gateway_v0::rotate_signers_data_hash(message_data);
+    let message_to_sign = bcs::to_bytes(&auth::new_message_to_sign(
+        domain_separator,
+        weighted_signers.hash(),
+        data_hash,
+    ));
+    let signature = axelar_gateway::proof::new_signature(
+        sui::ecdsa_k1::secp256k1_sign(keypair.private_key(), &message_to_sign, 0, true)
+    );
+    let proof = axelar_gateway::proof::create_for_testing(weighted_signers, vector[signature]);
```
Comment (milapsheth) [inline move/axelar_gateway/sources/gateway.move:786]: reuse

File: move/axelar_gateway/sources/gateway.move [hunk 2]
```diff
@@ -469,37 +564,474 @@ fun test_peel_proof_no_remaining_data() {
...
+
+    clock.increment_for_testing(minimum_rotation_delay);
+    self.rotate_signers(&clock, message_data, bcs::to_bytes(&proof), ctx);
+    
+    clock.destroy_for_testing();
+    sui::test_utils::destroy(self);
+}
+
+#[test]
+#[expected_failure]
```
Comment (milapsheth) [inline move/axelar_gateway/sources/gateway.move:796]: abort code

File: move/axelar_gateway/sources/gateway.move [hunk 3]
```diff
@@ -469,37 +564,474 @@ fun test_peel_proof_no_remaining_data() {
...
+fun test_rotate_signers_remaining_data_1() {
```
Comment (milapsheth) [inline move/axelar_gateway/sources/gateway.move:797]: avoid `_1`, `_2` naming. Make it more descriptive to the scenario

File: move/axelar_gateway/sources/gateway.move [hunk 4]
```diff
@@ -469,37 +564,474 @@ fun test_peel_proof_no_remaining_data() {
...
+    clock.increment_for_testing(minimum_rotation_delay);
+    self.rotate_signers(&clock, message_data, bcs::to_bytes(&proof), ctx);
+    
+    clock.destroy_for_testing();
+    sui::test_utils::destroy(self);
+}
+
+#[test]
+#[expected_failure]
+fun test_rotate_signers_remaining_data_2() {
```
Comment (milapsheth) [inline move/axelar_gateway/sources/gateway.move:859]: same

File: move/axelar_gateway/sources/gateway.move [hunk 5]
```diff
@@ -469,37 +564,474 @@ fun test_peel_proof_no_remaining_data() {
...
+        source_address,
+        destination_id,
+        payload_hash,
+    ), 0);
+        assert!(!gateway.is_message_executed(
+        source_chain,
+        message_id,
+    ), 1);
+
+
```
Comment (milapsheth) [inline move/axelar_gateway/sources/gateway.move:1015]:
```suggestion
```

File: move/axelar_gateway/sources/gateway.move [hunk 6]
```diff
@@ -430,6 +448,83 @@ fun test_setup() {
     scenario.end();
 }
 
+#[test]
+#[expected_failure]
```
[the remark this answers is not in the document]
Comment (Foivos) [reply move/axelar_gateway/sources/gateway.move:452]: utils aborts without an abort code.

File: move/axelar_gateway/sources/gateway.move [hunk 7]
```diff
@@ -430,6 +448,83 @@ fun test_setup() {
...
+    id.delete();
+
+    let (
+        operator_result,
+        messages,
+        signers,
+        _,
+    ) = inner.destroy<GatewayV0>().destroy_for_testing();
+
+    assert!(operator == operator_result, 1);
```
[the remark this answers is not in the document]
Comment (Foivos) [reply move/axelar_gateway/sources/gateway.move:502]: it does, but i like the abort codes for ease of debugging. I can remove them if you think it's ugly.

File: move/axelar_gateway/sources/gateway.move [hunk 8]
```diff
@@ -469,37 +564,474 @@ fun test_peel_proof_no_remaining_data() {
...
+    let proof = axelar_gateway::proof::create_for_testing(weighted_signers, vector[signature]);
+
+    self.approve_messages(message_data, bcs::to_bytes(&proof));
+    
+    clock.destroy_for_testing();
+    sui::test_utils::destroy(self)
+}
+
+#[test]
+#[expected_failure]
```
[the remark this answers is not in the document]
Comment (Foivos) [reply move/axelar_gateway/sources/gateway.move:667]: again, utils does not have an abort code.

File: move/axelar_gateway/sources/gateway.move (hunk 2 above)
Comment (Foivos) [reply move/axelar_gateway/sources/gateway.move:778]: again, utils does not have an abort code.

**Review by milapsheth — APPROVED** [review]

**Pull request merged**